spec/schema_tools/hash_spec.rb

```diff
@@ -16,7 +16,15 @@ class Contact
 end
 
 class OneOfDefinition
-  attr_accessor :person
+  attr_accessor :person, :one_of_schema, :all_of_schema, :any_of_schema
+end
+
+class AnyOfDefinition
+  attr_accessor :person, :one_of_schema, :all_of_schema, :any_of_schema
```

Comment:
on your comment above, this does not actually handle any of that, the gem wont stop you from using everything in an oneOf node or any of that (In fact, oneOf support was kind of already in this gem but it only worked if you had things at  a specific level in the json and thats how it behaved)